Lib/asyncio/events.py

```diff
@@ -101,7 +101,8 @@ class TimerHandle(Handle):
     __slots__ = ['_scheduled', '_when']
 
     def __init__(self, when, callback, args, loop, context=None):
-        assert when is not None
+        if when is None:
+            raise ValueError('when cannot be None')
```

Comment:
Would a `TypeError` be better here, since `NoneType` is a distinct type?

Reply:
I think ``ValueError`` is better as ``None`` is in value context here.